### PR TITLE
Back to top js file scroll-func.js improved

### LINKED ISSUE #7454: Displaying of go to top button
At the top of webstie the go to top button should not be displayed which is displaying in this case.

Screenshot from 2020-02-10 18-53-07

## Thank you!

Your help makes Public Lab better! We *deeply* appreciate your helping refine and improve this site. 

To learn how to write really great issues, which increases the chances they'll be resolved, see:

https://publiclab.org/wiki/developers#Contributing+for+non-coders

### PR DESCRIPTION
**Pull request by NitinBhasneria**
Fixes #7454 (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by NitinBhasneria** [comment]
@publiclab/reviewers @VladimirMikulic @nstjean Can you guide why this test failed ?

**Comment by VladimirMikulic** [comment]
The test failure is not your fault. It's the MySql connection error.

**Comment by VladimirMikulic** [comment]
I've just noticed that your branch's name is `f1`. You don't have to change it now, but in the future please name your branches so they have the actual meaning.

**Comment by NitinBhasneria** [comment]
Ok thanks @VladimirMikulic

**Comment by NitinBhasneria** [comment]
@VladimirMikulic Is there something to do for this error.

**Comment by VladimirMikulic** [comment]
@NitinBhasneria as I said, the workaround is to force push. MySql connection fails for some reason.

**Comment by SidharthBansal** [comment]
Thanks for collaboration here!

**Comment by NitinBhasneria** [comment]
@cesswairimu  @jywarren can you please approve and merge this.

**Comment by cesswairimu** [comment]
Hi @NitinBhasneria, I believe this was solved in https://github.com/publiclab/plots2/issues/6684 . The changes may not be in prod yet kindly check on stable.publiclab.org if its still an issue. Thanks

**Comment by NitinBhasneria** [comment]
Thanks @cesswairimu but at stable.publiclab.org I cant see any back to top button. It is totally removed there. Is it correct or should I make to appear it back.
It means there is no back to top button.
Screenshot from 2020-02-13 00-49-50

What are your views?
thnx

**Comment by cesswairimu** [comment]
Maybe it was never there :thinking:I can't see it locally either

**Comment by NitinBhasneria** [comment]
How will we know then is solved or not yet? @cesswairimu

**Comment by cesswairimu** [comment]
I will check if the changes on the other PR are in production

**Comment by cesswairimu** [comment]
How did you test the fix on this PR? Maybe that can give us pointers

**Comment by NitinBhasneria** [comment]
I tested the function in scroll-func.js by giving the CSS output in the console.
@cesswairimu

**Comment by NitinBhasneria** [comment]
@nstjean Can you check this please.

**Comment by nstjean** [comment]
I can't find the code anywhere in the repository. I took a look at it on the live site and I'm pretty sure that that code is included in the footer feature - so it only shows up on the live site, not stable and not local. That makes it harder to test!

**Comment by nstjean** [comment]
Also it does appear that this code already exists here:
https://github.com/publiclab/plots2/blob/5c0d96274bd0be15feba69d8f30093082402ace5/app/assets/javascripts/header_footer.js

**Comment by NitinBhasneria** [comment]
Looks like the issue is already solved then.

**Comment by nstjean** [comment]
But does the existing javascript hide the arrow at the top of the page?

**Comment by NitinBhasneria** [comment]
If the code link sent by you is the only for the arrow(back to top) then it doesn't hide.
How can we check?

**Comment by nstjean** [comment]
What I would do is copy the code from the live site into views/layouts/_footer.html.erb just for testing.

**Comment by nstjean** [comment]
```
<a href="#" class="back-to-top" style="display: inline;">
   <i class="fa fa-arrow-circle-up"></i>
</a>
```

**Comment by NitinBhasneria** [comment]
Good idea! 
I will test with this.
thanks @nstjean

**Comment by NitinBhasneria** [comment]
@nstjean As per your guidance I have seen the back to top function but It has not been solved yet. You can see this in this screenshot.
Screenshot from 2020-03-09 19-40-02

File: app/assets/javascripts/scroll-func.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+let scrollBtn = document.getElementsByClassName('back-to-top');
+
+window.addEventListener('scroll', function() {
+  let position = window.scrollY;
+
+  if (position < 200) {
+    scrollBtn.header.style.opacity = '0';
```
Comment (VladimirMikulic) [inline app/assets/javascripts/scroll-func.js]: This should be `scrollBtn.style.display="none";`
@NitinBhasneria that's why this doesn't work.

File: app/assets/javascripts/scroll-func.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  } else {
+    scrollBtn.header.style.opacity = '1';
```
Comment (VladimirMikulic) [inline app/assets/javascripts/scroll-func.js]: Adjust this line as well.

**Comment by VladimirMikulic** [comment]
To avoid performance issues on the site, we could also make a debouncer to ignore "scroll" events.
Firing the function every time user scrolls quickly creates a bottleneck.

**Comment by nstjean** [comment]
> To avoid performance issues on the site, we could also make a debouncer to ignore "scroll" events.
> Firing the function every time user scrolls quickly creates a bottleneck.

True! Here is some good information for anyone who is interested: https://www.sitepoint.com/throttle-scroll-events/

**Comment by cesswairimu** [comment]
restarting travis

**Comment by jywarren** [comment]
Hi all! Noting that this button is currently in the `footer-notice` feature in the database (https://github.com/publiclab/plots2/blob/main/doc/DATA_MODEL.md#features) on production. I'll add it to stable too, to match!

I hope that will help debugging here! cc @fiona-byte @cesswairimu 

Update: looks like it does exist in stable, and also i noticed i am seeing the button!

image

**Pull request closed without merging**